### PR TITLE
Add sample to load data to gsheet

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: load-data-to-gsheet/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "lakshanweerasinghe"
```
Comment (SasinduDilshara) [inline load-data-to-gsheet/Ballerina.toml]: Shall we add a `org` name like `ballerina`.

File: load-data-to-gsheet/main.bal [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+public type ItemData record {|
+    string salesId;
+    string productId;
+    float unitPrice;
+    float quantity;
+|};
+
+configurable string refreshToken = ?;
+configurable string clientId = ?;
+configurable string clientSecret = ?;
```
Comment (SasinduDilshara) [inline load-data-to-gsheet/main.bal:5]: imo, config variables should be top of the file. wdyt?

File: load-data-to-gsheet/main.bal [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    // create a new spread sheet to load data
+    sheets:Spreadsheet spreadsheet = check spreadsheetClient->createSpreadsheet(sheetName);
+    string spreadSheetId = spreadsheet.spreadsheetId;
+
+    // add legends to the sheet
+    _ = check spreadsheetClient->appendValue(spreadSheetId, ["Product", "Sales"], {sheetName: sheetName});
+    foreach var {productId, sales} in salesSummary {
+        // add values to the sheet
+        _ = check spreadsheetClient->appendValue(
+            spreadSheetId, [productId, sales], {sheetName});
```
Comment (SasinduDilshara) [inline load-data-to-gsheet/main.bal]:
```suggestion
        _ = check spreadsheetClient->appendValue(spreadSheetId, [productId, sales], {sheetName});
```

File: load-data-to-gsheet/main.bal [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+}
+
+public function main() returns error? {
+    ItemData[] salesData = [];
+    SalesSummary[] salesSummary = aggregateSalesData(salesData);
```
Comment (SasinduDilshara) [inline load-data-to-gsheet/main.bal]:
```suggestion
    SalesSummary[] salesSummary = aggregateSalesData([]);
```
wdyt?

File: load-data-to-gsheet/Ballerina.toml (hunk 1 above)
Comment (LakshanWeerasinghe) [reply load-data-to-gsheet/Ballerina.toml]: I think it should be something like `ballerinaetl`

File: load-data-to-gsheet/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+```
+refreshToken = <REFRESH_TOKEN>
+clientId = <CLIENT-ID>
+clientSecret = <CLIENT-SECRET>
+```
+
+### Running the sample
+Run the following command from the root directory of the sample.
+```ballerina
+bal run
```
Comment (SasinduDilshara) [inline load-data-to-gsheet/README.md]:
```suggestion
$ bal run
```

File: load-data-to-gsheet/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+## Steps to run the sample
+
+### Obtaining Google sheets credentials
+1. Create a [Google account](https://accounts.google.com/signup/v2/createaccount)
+2. Obtain your tokens using this [link](https://developers.google.com/identity/protocols/oauth2) and include it in the `Config.toml` file.
+
+### Updating the Config.toml file
+```
+refreshToken = <REFRESH_TOKEN>
+clientId = <CLIENT-ID>
```
Comment (SasinduDilshara) [inline load-data-to-gsheet/README.md]:
```suggestion
clientId = <CLIENT_ID>
```

**Review by SasinduDilshara — DISMISSED** [review]

File: load-data-to-gsheet/Ballerina.toml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "etl"
+name = "load_data_to_gsheet"
+version = "0.1.0"
+distribution = "2201.8.2"
+
+[build-options]
+observabilityIncluded = true
```
Comment (pcnfernando) [inline load-data-to-gsheet/Ballerina.toml]: Can remove this

File: load-data-to-gsheet/main.bal [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import ballerinax/googleapis.sheets as sheets;
```
Comment (pcnfernando) [inline load-data-to-gsheet/main.bal]: You don't need the alias right?

File: load-data-to-gsheet/main.bal [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+final sheets:Client spreadsheetClient = check new ({
+    auth: {
+        clientId,
+        clientSecret,
+        refreshUrl: sheets:REFRESH_URL,
+        refreshToken
+    }
+});
+
+public function loadToGoogleSheet(string sheetName, string workSheetName, SalesSummary[] salesSummary) returns error? {
```
Comment (pcnfernando) [inline load-data-to-gsheet/main.bal]:
```suggestion
function loadToGoogleSheet(string sheetName, string workSheetName, SalesSummary[] salesSummary) returns error? {
```

File: load-data-to-gsheet/main.bal (hunk 8 above)
Comment (LakshanWeerasinghe) [reply load-data-to-gsheet/main.bal]: Yes, we can remove it.

File: load-data-to-gsheet/README.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
+
+## Steps to run the sample
+
+### Obtaining Google sheets credentials
```
Comment (pcnfernando) [inline load-data-to-gsheet/README.md]: Use the same convension as in other samples

Comment (LakshanWeerasinghe) [reply load-data-to-gsheet/README.md]: Updated please check.

Comment (LakshanWeerasinghe) [reply load-data-to-gsheet/README.md]: Updated in [139ba](https://github.com/ballerina-guides/etl-samples/pull/8/commits/139bab7a1b37c1f9ef4c163e1c7e23fd5b6e2ed5).

**Review by pcnfernando — APPROVED** [review]
Suggestions will be handled separately. Hence approving

**Pull request merged**